src/main/java/org/testng/internal/TestResult.java

```diff
@@ -30,6 +31,7 @@
   private long m_startMillis = 0;
   private long m_endMillis = 0;
   private String m_name = null;
+  private String m_alteredName = null;

```

Comment:
I will try this. I remember seeing some awkward behavior when i tried to update `m_name` directly via the new interface implementation. Can't remember it (vices of writing code early in the morning when you are still half awake :p )